database/tests/authentication_tests/test_tableau_api_auth.py

```diff
@@ -90,38 +126,92 @@ def test_check_permissions_none(self):
 
     def test_check_permissions_inactive(self):
         self.setup(researcher=True, apikey=True, study=True)
-        ApiKey.objects.filter(access_key_id=self.api_key_public).update(is_active=False)
-        headers = {"X-Access-Key-Id": self.api_key_public, "X-Access-Key-Secret": self.api_key_private}
+        self.api_key.update(is_active=False)
+        headers = {
+            "X-Access-Key-Id": self.api_key_public,
+            "X-Access-Key-Secret": self.api_key_private,
+        }
         with self.assertRaises(AuthenticationFailed) as cm:
             with app.test_request_context(headers=headers):
                 TableauApiView().check_permissions(study_id=self.study.object_id)
 
     def test_check_permissions_bad_secret(self):
         self.setup(researcher=True, apikey=True, study=True)
-        headers = {"X-Access-Key-Id": self.api_key_public, "X-Access-Key-Secret": ':::' + self.api_key_private[3:]}
+        # note that ':' does not appear in base64 encoding, preventing any collision errors based on the current
+        # implementation
+        headers = {
+            "X-Access-Key-Id": self.api_key_public,
+            "X-Access-Key-Secret": ":::" + self.api_key_private[3:],
+        }
         with self.assertRaises(AuthenticationFailed) as cm:
             with app.test_request_context(headers=headers):
                 TableauApiView().check_permissions(study_id=self.study.object_id)
 
     def test_check_permissions_no_tableau(self):
         self.setup(researcher=True, apikey=True, study=True)
-        ApiKey.objects.filter(access_key_id=self.api_key_public).update(has_tableau_api_permissions=False)
-        headers = {"X-Access-Key-Id": self.api_key_public, "X-Access-Key-Secret": self.api_key_private}
+        ApiKey.objects.filter(access_key_id=self.api_key_public).update(
+            has_tableau_api_permissions=False
+        )
+        headers = {
+            "X-Access-Key-Id": self.api_key_public,
+            "X-Access-Key-Secret": self.api_key_private,
+        }
         with self.assertRaises(PermissionDenied) as cm:
             with app.test_request_context(headers=headers):
                 TableauApiView().check_permissions(study_id=self.study.object_id)
 
     def test_check_permissions_bad_study(self):
         self.setup(researcher=True, apikey=True, study=True)
-        headers = {"X-Access-Key-Id": self.api_key_public, "X-Access-Key-Secret": self.api_key_private}
+        headers = {
+            "X-Access-Key-Id": self.api_key_public,
+            "X-Access-Key-Secret": self.api_key_private,
+        }
+        self.assertFalse(ApiKey.objects.filter(access_key_id=" bad study id ").exists())
         with self.assertRaises(PermissionDenied) as cm:
             with app.test_request_context(headers=headers):
                 TableauApiView().check_permissions(study_id=" bad study id ")
```

Comment:
1. I think you're looking for `Study.objects.filter(object_id=" bad study id ")`? I wrote the wrong one in my comment haha.
2. Probably worth factoring out `" bad study id "` into its own variable